File: binding/include/multimedia/libqimagecapture.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qimagecapture.html
pub const qimagecapture = struct {

    /// New constructs a new QImageCapture object.
    pub fn New() QtC.QImageCapture {
        return qtc.QImageCapture_new();
    }


    /// New2 constructs a new QImageCapture object.
    pub fn New2(parent: ?*anyopaque) QtC.QImageCapture {
        return qtc.QImageCapture_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QImageCapture_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qimagecapture.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn IsAvailable(self: ?*anyopaque, ) bool {
        return qtc.QImageCapture_IsAvailable(@ptrCast(self));
    }

    pub fn CaptureSession(self: ?*anyopaque, ) QtC.QMediaCaptureSession {
        return qtc.QImageCapture_CaptureSession(@ptrCast(self));
    }

    pub fn Error(self: ?*anyopaque, ) i32 {
        return qtc.QImageCapture_Error(@ptrCast(self));
    }

    pub fn ErrorString(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QImageCapture_ErrorString(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qimagecapture.ErrorString: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn IsReadyForCapture(self: ?*anyopaque, ) bool {
        return qtc.QImageCapture_IsReadyForCapture(@ptrCast(self));
    }

    pub fn FileFormat(self: ?*anyopaque, ) i32 {
        return qtc.QImageCapture_FileFormat(@ptrCast(self));
    }

    pub fn SetFileFormat(self: ?*anyopaque, format: i32) void {
        qtc.QImageCapture_SetFileFormat(@ptrCast(self), @intCast(format));
    }

    pub fn SupportedFormats() []const u8 {
        return qtc.QImageCapture_SupportedFormats();
    }

    pub fn FileFormatName(c: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QImageCapture_FileFormatName(@intCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qimagecapture.FileFormatName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn FileFormatDescription(c: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QImageCapture_FileFormatDescription(@intCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qimagecapture.FileFormatDescription: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Resolution(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QImageCapture_Resolution(@ptrCast(self));
    }

    pub fn SetResolution(self: ?*anyopaque, resolution: ?*anyopaque) void {
        qtc.QImageCapture_SetResolution(@ptrCast(self), @ptrCast(resolution));
    }

    pub fn SetResolution2(self: ?*anyopaque, width: i32, height: i32) void {
        qtc.QImageCapture_SetResolution2(@ptrCast(self), width, height);
    }

    pub fn Quality(self: ?*anyopaque, ) i32 {
        return qtc.QImageCapture_Quality(@ptrCast(self));
    }

    pub fn SetQuality(self: ?*anyopaque, quality: i32) void {
        qtc.QImageCapture_SetQuality(@ptrCast(self), @intCast(quality));
    }

    pub fn MetaData(self: ?*anyopaque, ) QtC.QMediaMetaData {
        return qtc.QImageCapture_MetaData(@ptrCast(self));
    }

    pub fn SetMetaData(self: ?*anyopaque, metaData: ?*anyopaque) void {
        qtc.QImageCapture_SetMetaData(@ptrCast(self), @ptrCast(metaData));
    }

    pub fn AddMetaData(self: ?*anyopaque, metaData: ?*anyopaque) void {
        qtc.QImageCapture_AddMetaData(@ptrCast(self), @ptrCast(metaData));
    }

    pub fn CaptureToFile(self: ?*anyopaque, ) i32 {
        return qtc.QImageCapture_CaptureToFile(@ptrCast(self));
    }

    pub fn Capture(self: ?*anyopaque, ) i32 {
        return qtc.QImageCapture_Capture(@ptrCast(self));
    }

    pub fn ErrorChanged(self: ?*anyopaque, ) void {
        qtc.QImageCapture_ErrorChanged(@ptrCast(self));
    }

    pub fn OnErrorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ErrorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ErrorOccurred(self: ?*anyopaque, id: i32, errorVal: i32, errorString: []const u8) void {
        const errorString_str = qtc.libqt_string{
    .len = errorString.len,
    .data = errorString.ptr,
};
qtc.QImageCapture_ErrorOccurred(@ptrCast(self), id, @intCast(errorVal), errorString_str);
    }

    pub fn OnErrorOccurred(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, i32, [*:0]const u8) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ErrorOccurred(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ReadyForCaptureChanged(self: ?*anyopaque, ready: bool) void {
        qtc.QImageCapture_ReadyForCaptureChanged(@ptrCast(self), ready);
    }

    pub fn OnReadyForCaptureChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ReadyForCaptureChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn MetaDataChanged(self: ?*anyopaque, ) void {
        qtc.QImageCapture_MetaDataChanged(@ptrCast(self));
    }

    pub fn OnMetaDataChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_MetaDataChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FileFormatChanged(self: ?*anyopaque, ) void {
        qtc.QImageCapture_FileFormatChanged(@ptrCast(self));
    }

    pub fn OnFileFormatChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_FileFormatChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn QualityChanged(self: ?*anyopaque, ) void {
        qtc.QImageCapture_QualityChanged(@ptrCast(self));
    }

    pub fn OnQualityChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_QualityChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ResolutionChanged(self: ?*anyopaque, ) void {
        qtc.QImageCapture_ResolutionChanged(@ptrCast(self));
    }

    pub fn OnResolutionChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ResolutionChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ImageExposed(self: ?*anyopaque, id: i32) void {
        qtc.QImageCapture_ImageExposed(@ptrCast(self), id);
    }

    pub fn OnImageExposed(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ImageExposed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ImageCaptured(self: ?*anyopaque, id: i32, preview: ?*anyopaque) void {
        qtc.QImageCapture_ImageCaptured(@ptrCast(self), id, @ptrCast(preview));
    }

    pub fn OnImageCaptured(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, ?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ImageCaptured(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ImageMetadataAvailable(self: ?*anyopaque, id: i32, metaData: ?*anyopaque) void {
        qtc.QImageCapture_ImageMetadataAvailable(@ptrCast(self), id, @ptrCast(metaData));
    }

    pub fn OnImageMetadataAvailable(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, ?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ImageMetadataAvailable(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ImageAvailable(self: ?*anyopaque, id: i32, frame: ?*anyopaque) void {
        qtc.QImageCapture_ImageAvailable(@ptrCast(self), id, @ptrCast(frame));
    }

    pub fn OnImageAvailable(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, ?*anyopaque) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ImageAvailable(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ImageSaved(self: ?*anyopaque, id: i32, fileName: []const u8) void {
        const fileName_str = qtc.libqt_string{
    .len = fileName.len,
    .data = fileName.ptr,
};
qtc.QImageCapture_ImageSaved(@ptrCast(self), id, fileName_str);
    }

    pub fn OnImageSaved(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, [*:0]const u8) callconv(.c) void) void {
        qtc.QImageCapture_Connect_ImageSaved(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QImageCapture_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qimagecapture.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QImageCapture_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qimagecapture.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn CaptureToFile1(self: ?*anyopaque, location: []const u8) i32 {
        const location_str = qtc.libqt_string{
    .len = location.len,
    .data = location.ptr,
};
return qtc.QImageCapture_CaptureToFile1(@ptrCast(self), location_str);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QImageCapture_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/imagecapture.html#types
pub const enums = struct {
    pub const Error = enum {
        pub const NoError: i32 = 0;
        pub const NotReadyError: i32 = 1;
        pub const ResourceError: i32 = 2;
        pub const OutOfSpaceError: i32 = 3;
        pub const NotSupportedFeatureError: i32 = 4;
        pub const FormatError: i32 = 5;
    };

    pub const Quality = enum {
        pub const VeryLowQuality: i32 = 0;
        pub const LowQuality: i32 = 1;
        pub const NormalQuality: i32 = 2;
        pub const HighQuality: i32 = 3;
        pub const VeryHighQuality: i32 = 4;
    };

    pub const FileFormat = enum {
        pub const UnspecifiedFormat: i32 = 0;
        pub const JPEG: i32 = 1;
        pub const PNG: i32 = 2;
        pub const WebP: i32 = 3;
        pub const Tiff: i32 = 4;
        pub const LastFileFormat: i32 = 4;
    };

};

File: binding/src/multimedia/libqimagecapture.cpp
#include <QImage>
#include <QImageCapture>
#include <QMediaCaptureSession>
#include <QMediaMetaData>
#include <QObject>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QVideoFrame>
#include <qimagecapture.h>
#include "libqimagecapture.h"
#include "libqimagecapture.hxx"

QImageCapture* QImageCapture_new() {
	 return new QImageCapture();
}

QImageCapture* QImageCapture_new2(QObject* parent) {
	 return new QImageCapture(parent);
}

libqt_string QImageCapture_Tr(const char* s) {
	QString _ret = QImageCapture::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QImageCapture_IsAvailable(const QImageCapture* self) {
	return self->isAvailable();
}

QMediaCaptureSession* QImageCapture_CaptureSession(const QImageCapture* self) {
	return self->captureSession();
}

int QImageCapture_Error(const QImageCapture* self) {
	return self->error();
}

libqt_string QImageCapture_ErrorString(const QImageCapture* self) {
	QString _ret = self->errorString();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QImageCapture_IsReadyForCapture(const QImageCapture* self) {
	return self->isReadyForCapture();
}

int QImageCapture_FileFormat(const QImageCapture* self) {
	return self->fileFormat();
}

void QImageCapture_SetFileFormat(QImageCapture* self, int format) {
	self->setFileFormat(static_cast<QMediaFormat::FileFormat>(format));
}

libqt_list QImageCapture_SupportedFormats() {
	auto _ret = QImageCapture::supportedFormats();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		auto& _elem = _ret[_i];
		_data[_i] = new std::remove_reference_t<decltype(_elem)>(_elem);
	}
	return _arr;
}

libqt_string QImageCapture_FileFormatName(int c) {
	QString _ret = QImageCapture::fileFormatName(static_cast<QMediaFormat::FileFormat>(c));
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QImageCapture_FileFormatDescription(int c) {
	QString _ret = QImageCapture::fileFormatDescription(static_cast<QMediaFormat::FileFormat>(c));
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QSize* QImageCapture_Resolution(const QImageCapture* self) {
	return new QSize(self->resolution());
}

void QImageCapture_SetResolution(QImageCapture* self, const QSize* resolution) {
	self->setResolution(*resolution);
}

void QImageCapture_SetResolution2(QImageCapture* self, int width, int height) {
	self->setResolution(width, height);
}

int QImageCapture_Quality(const QImageCapture* self) {
	return self->quality();
}

void QImageCapture_SetQuality(QImageCapture* self, int quality) {
	self->setQuality(static_cast<QMediaRecorder::Quality>(quality));
}

QMediaMetaData* QImageCapture_MetaData(const QImageCapture* self) {
	return new QMediaMetaData(self->metaData());
}

void QImageCapture_SetMetaData(QImageCapture* self, const QMediaMetaData* metaData) {
	self->setMetaData(*metaData);
}

void QImageCapture_AddMetaData(QImageCapture* self, const QMediaMetaData* metaData) {
	self->addMetaData(*metaData);
}

int QImageCapture_CaptureToFile(QImageCapture* self) {
	return self->captureToFile();
}

int QImageCapture_Capture(QImageCapture* self) {
	return self->capture();
}

void QImageCapture_ErrorChanged(QImageCapture* self) {
	self->errorChanged();
}

void QImageCapture_Connect_ErrorChanged(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*) = reinterpret_cast<void (*)(QImageCapture*)>(slot);
    QImageCapture::connect(self, &QImageCapture::errorChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QImageCapture_ErrorOccurred(QImageCapture* self, int id, int errorVal, const libqt_string errorString) {
	self->errorOccurred(id, static_cast<QImageCapture::Error>(errorVal), QString::fromUtf8(errorString.data, errorString.len));
}

void QImageCapture_Connect_ErrorOccurred(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*, int, int, const char*) = reinterpret_cast<void (*)(QImageCapture*, int, int, const char*)>(slot);
    QImageCapture::connect(self, &QImageCapture::errorOccurred, [self, slotFunc](int id, QImageCapture::Error errorVal, const QString& errorString) {
	slotFunc(self, id, errorVal, errorString);
    });
}

void QImageCapture_ReadyForCaptureChanged(QImageCapture* self, bool ready) {
	self->readyForCaptureChanged(ready);
}

void QImageCapture_Connect_ReadyForCaptureChanged(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*, bool) = reinterpret_cast<void (*)(QImageCapture*, bool)>(slot);
    QImageCapture::connect(self, &QImageCapture::readyForCaptureChanged, [self, slotFunc](bool ready) {
	slotFunc(self, ready);
    });
}

void QImageCapture_MetaDataChanged(QImageCapture* self) {
	self->metaDataChanged();
}

void QImageCapture_Connect_MetaDataChanged(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*) = reinterpret_cast<void (*)(QImageCapture*)>(slot);
    QImageCapture::connect(self, &QImageCapture::metaDataChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QImageCapture_FileFormatChanged(QImageCapture* self) {
	self->fileFormatChanged();
}

void QImageCapture_Connect_FileFormatChanged(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*) = reinterpret_cast<void (*)(QImageCapture*)>(slot);
    QImageCapture::connect(self, &QImageCapture::fileFormatChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QImageCapture_QualityChanged(QImageCapture* self) {
	self->qualityChanged();
}

void QImageCapture_Connect_QualityChanged(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*) = reinterpret_cast<void (*)(QImageCapture*)>(slot);
    QImageCapture::connect(self, &QImageCapture::qualityChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QImageCapture_ResolutionChanged(QImageCapture* self) {
	self->resolutionChanged();
}

void QImageCapture_Connect_ResolutionChanged(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*) = reinterpret_cast<void (*)(QImageCapture*)>(slot);
    QImageCapture::connect(self, &QImageCapture::resolutionChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QImageCapture_ImageExposed(QImageCapture* self, int id) {
	self->imageExposed(id);
}

void QImageCapture_Connect_ImageExposed(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*, int) = reinterpret_cast<void (*)(QImageCapture*, int)>(slot);
    QImageCapture::connect(self, &QImageCapture::imageExposed, [self, slotFunc](int id) {
	slotFunc(self, id);
    });
}

void QImageCapture_ImageCaptured(QImageCapture* self, int id, const QImage* preview) {
	self->imageCaptured(id, *preview);
}

void QImageCapture_Connect_ImageCaptured(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*, int, const QImage*) = reinterpret_cast<void (*)(QImageCapture*, int, const QImage*)>(slot);
    QImageCapture::connect(self, &QImageCapture::imageCaptured, [self, slotFunc](int id, const QImage& preview) {
	slotFunc(self, id, preview);
    });
}

void QImageCapture_ImageMetadataAvailable(QImageCapture* self, int id, const QMediaMetaData* metaData) {
	self->imageMetadataAvailable(id, *metaData);
}

void QImageCapture_Connect_ImageMetadataAvailable(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*, int, const QMediaMetaData*) = reinterpret_cast<void (*)(QImageCapture*, int, const QMediaMetaData*)>(slot);
    QImageCapture::connect(self, &QImageCapture::imageMetadataAvailable, [self, slotFunc](int id, const QMediaMetaData& metaData) {
	slotFunc(self, id, metaData);
    });
}

void QImageCapture_ImageAvailable(QImageCapture* self, int id, const QVideoFrame* frame) {
	self->imageAvailable(id, *frame);
}

void QImageCapture_Connect_ImageAvailable(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*, int, const QVideoFrame*) = reinterpret_cast<void (*)(QImageCapture*, int, const QVideoFrame*)>(slot);
    QImageCapture::connect(self, &QImageCapture::imageAvailable, [self, slotFunc](int id, const QVideoFrame& frame) {
	slotFunc(self, id, frame);
    });
}

void QImageCapture_ImageSaved(QImageCapture* self, int id, const libqt_string fileName) {
	self->imageSaved(id, QString::fromUtf8(fileName.data, fileName.len));
}

void QImageCapture_Connect_ImageSaved(QImageCapture* self, intptr_t slot) {
    void (*slotFunc)(QImageCapture*, int, const char*) = reinterpret_cast<void (*)(QImageCapture*, int, const char*)>(slot);
    QImageCapture::connect(self, &QImageCapture::imageSaved, [self, slotFunc](int id, const QString& fileName) {
	slotFunc(self, id, fileName);
    });
}

libqt_string QImageCapture_Tr2(const char* s, const char* c) {
	QString _ret = QImageCapture::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QImageCapture_Tr3(const char* s, const char* c, int n) {
	QString _ret = QImageCapture::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QImageCapture_CaptureToFile1(QImageCapture* self, const libqt_string location) {
	return self->captureToFile(QString::fromUtf8(location.data, location.len));
}

void QImageCapture_Delete(QImageCapture* self) {
    delete self;
}

